### PR TITLE
Add neuralynx interface

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@bendichter Basic addition of Neuralynx interface

Though it doesn't yet wok for the brody-lab-to-nwb project ([but issue posted on neo](https://github.com/NeuralEnsemble/python-neo/issues/959)) it should support any other type of `.ncs` files of the appropriate version (that's one of the ones the GIN tests are passing for).

### CODE REVIEW

**Comment by bendichter** [comment]
@CodyCBakerPhD do you know what is causing the CI to fail?

**Comment by CodyCBakerPhD** [comment]
@bendichter Ahhh yes - I'll have to mimic the behavior of other interfaces dependent on neo extractors (they don't have method signatures the way standard extractors do). Fixing now

**Comment by CodyCBakerPhD** [comment]
Well, turns out this will require a modification on spikeextractors - Luiz got the blackrock one to work because he overrode the init just to attach the annotation type checking. Changing this to draft until spikeextractors is modified

**Comment by bendichter** [comment]
You should be able to just override `get_source_schema` for this class

**Comment by bendichter** [comment]
which I see you have done. Why isn't that working?

**Comment by CodyCBakerPhD** [comment]
Yeah, so what I've overridden here is @luiztauffer adaptation of the `get_method_from_signature` to add in the extra properties/descriptions but it still relies on the annotations in the `__init__` that [spikeextractors does not have for Neuralynx](https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/neoextractors/neuralynxextractor.py), but does have from what [Luiz did on blackrock](https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/neoextractors/blackrockextractor.py).

Now, I could further override the source_schema here temporarily to simply construct the entire thing manually  until it gets added in spikeextractors - BUT I caution against this since [one potential solution](https://github.com/NeuralEnsemble/python-neo/pull/957) for making the neuralynx usage more robust in neo (required for brody-lab) requires different arguments altogether.

**Comment by CodyCBakerPhD** [comment]
Progress being made on the neo performance issues with [neo-PR#990](https://github.com/NeuralEnsemble/python-neo/pull/990). Can finish up this PR once those are through.

File: nwb_conversion_tools/datainterfaces/neuralynxdatainterface.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                ),
+            ),
+            type="object",
+            additionalProperties=True
+        )
+
+    # The Neuralynx IO requires a slight workaround due to the folder based functionality of the neo IO being
+    # buggy with all the different versions of Neuralynx available. The safest method is to construct a
+    # MultiChannelRecordingExtractor containing individually loaded NeuralynxRecordingExtractors via filenames
+    def __init__(self, **source_data):
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/neuralynxdatainterface.py:34]:
```suggestion
    def __init__(self, dir_name: str):
```

File: nwb_conversion_tools/datainterfaces/neuralynxdatainterface.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        self.source_data = source_data
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/neuralynxdatainterface.py:35]:
```suggestion
        self.source_data = dict(dir_name=dir_name)
```

File: nwb_conversion_tools/datainterfaces/neuralynxdatainterface.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        neuralynx_files = [x for x in Path(self.source_data["dirname"]).iterdir() if ".ncs" in x.suffixes]
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/neuralynxdatainterface.py:36]:
```suggestion
        neuralynx_files = [x for x in Path(self.dirname).iterdir() if ".ncs" in x.suffixes]
```

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+h5py==2.10.0
+PyYAML==5.3.1
+# spikeextractors==0.9.3
+spikeextractors==0.9.5
+spikesorters==0.3.0
+spiketoolkit==0.6.1
+psutil==5.7.0
+git+https://github.com/catalystneuro/roiextractors
+neo
+=======
```
Comment (bendichter) [inline requirements.txt:17]: can you fix this?

Comment (CodyCBakerPhD) [reply requirements.txt:17]: Once PR #228 introduces the GIN framework, I'll add the Neuralynx tests and update this branch in the other respects to ensure it all works well

**Comment by CodyCBakerPhD** [comment]
Superseded by #235

**Pull request closed without merging**